vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -36,6 +36,11 @@ var DiskControllerTypes = []string{
 	"ide",
 }
 
+var NetworkAdapterTypes = []string{
```

Comment:
This variable should be unexported (starts with a lowercase letter, ensures that no package outside of `vsphere` can use it - basically a private variable), and be specific enough to avoid collisions.

Suggestion: `virtualMachineNetworkAdapterTypeAllowedValues`.

Also, you might want to check [here](https://godoc.org/github.com/vmware/govmomi/vim25/types) to see if there is any appropriate constants you can use.